### DISCUSSION TITLE
Calculation of LAI

### DISCUSSION THREAD

**Discussion by ouroukhai** [r-lidar/lidR#706] [Q&A] [answered]
Hi,

I came across this article https://doi.org/10.1016/j.rse.2014.10.004, and also read the documentation of leafR library about the lai() calculation. Seems like to calculate lai the following procedure should be followed:

normalized las ---> lad.voxels() ---> lad.profile() ---> lai()

To include such a calculation into a my_metrics function used in a pixel_metrics catalog handle (or any other *_metrics), is a bit weird as Zs are not exactly a lasfile. Is there any suggested way to calculate lai that could be included in a custom my_metrics function? 
My current solution is to include the calculation of LAI directly from equations (5) and (6) of the mentioned paper, but I was interested in any suggestion using the current dependencies.

Thank you in advance

**Comment by Jean-Romain** [comment]
The LAI is not a metric. It is a profile i.e. a serie of numbers. You must derive a number of that profile. In the article mentioned they use the coefficient of variation

**Comment by bi0m3trics** [comment, marked as answer]
Here's a nice review by Wang & Fang 2020 (https://www.mdpi.com/2072-4292/12/20/3457) that covers approaches commonly used to estimate LAI from ALS if you haven't seen it.

Personally, I really like figure 4 from Almeida et al. 2019 (https://www.mdpi.com/2072-4292/11/1/92) which visually explains how one might estimate LAI from LAD within a pixel (as a flattened column of voxels)
![image](https://github.com/r-lidar/lidR/assets/3956719/71e5cc9f-686d-41e6-8cce-470c027bf173)
and while it's been my experience that the resulting values (w/in the 2D unit area) slightly differ from LAI measured using field methods, it provides similar inferences and behaves similarly when included in models. 

Here's how I've been calculating it (e.g., in a 1m pixel) for ponderosa pine forests: 
`pixelsLAI = grid_metrics(las, ~as.numeric(sum(LAD(Z, dz = 1, k = 1.0, z0 = 1)$lad), na.rm = TRUE), 1)`
Disclaimer: Your parameters will be different in your system and you should explore the implications of that fully and it should be noted that I handle NAs and Zeros separately later (thus the na.rm = TRUE).

**Reply by ouroukhai** [reply]
This is very helpful, thank you very much! @bi0m3trics

**Reply by bi0m3trics** [reply]
Two more thoughts that came just came to me:

1. Be careful with the grain for the calculation (1m vs 10m vs 30m). LAI derived from LAD may not make sense at all scales.
2. I suggest having a look at [canopyLazR](https://github.com/akamoske/canopyLazR) in addition to [leafR](https://cran.r-project.org/web/packages/leafR/index.html). As always, there's more than one way to do things.
